lib/tasks/journeys_report.rake

```diff
@@ -64,15 +102,15 @@ namespace :journeys do
         file.write(JSON.pretty_generate({ moves: moves })) # maybe change to .to_json for production
       end
 
-      puts "Exporting #{filename} to S3..."
-      bucket_name = ENV['S3_REPORTING_BUCKET_NAME']
       if bucket_name.present?
+        puts "Exporting #{filename} to S3..."
         s3 = Aws::S3::Resource.new
         obj = s3.bucket(bucket_name).object(filename)
         obj.upload_file(filename)
       else
         puts 'S3_REPORTING_BUCKET_NAME not defined, skipping upload.'
       end
     end
+    update_bookmark(bucket_name)
```

Comment:
1. I think you should use the `timestamp` to label the bucket, as that is the time the process started, not the time when it finished.
2. you should only update the bookmark if you have uploaded files to S3, so the logic needs to change a little

Reply:
yes, those are very good points.   
It may take some time to run with large datasets.

Reply:
Not using a dedicated object for the bookmark.  Went with an approach between our timestamps and a typical S3 data partitioning format with a key prefix
but instead of:
```
PRE year=2020/month=06/day=01/
PRE year=2020/month=06/day=02/
```
I'm using a rather simpler
```
       PRE 1593773385/
       PRE 1593773505/
       PRE 1593773530/
```